### ISSUE TITLE
When we are using recurring pattern as - Every last day (suppose Wednesday) of months for a year starting of month having 5 week and our chosen day also lies in that week. SPParseEvent returns instances for recurring meeting  for only those months which have 5 weeks,

### ISSUE THREAD

**Issue by AAnkitaAswal** [JoeHogan/sharepoint-events-parser#3]
i had created a meeting which occurs every month on last Wednesday starting from Januray 2019 to December 2019 so 12 instances of this recurring meeting should be created.

So when this meeting was created in outlook, 12 instances of this meeting was created in SharePoint calendar list.
But when we are using SPEventParser to get all instances of this meeting using JS file in an content editor webpart we are not getting all instances .
You can see from screenshot for the same meeting all instances are not coming in our page, though all instances exist in SP List and outlook.
Inference:
We analyzed that only if we are taking start month of the invite which has move 5 weeks, SPEventParser is blinding looking for 5 week and skipping months which have 4 weeks that is why timeline in page is showing only those months which have 5 weeks (one exception is there is any months has 4 weeks but last week ends before the day of meeting, it is also coming in timeline).
Though it should not be case it should pick last day specified in meeting invite irrespective of weeks in that month.
Please let me know in case of further assistance is required in analyzing.

**Comment by shaipetel** [comment]
Similarly, when you have monthly event for first tuesday of every month it will only show up for months that begins before or on tuesday. So if your month begins on a wednesday, it won't show up on the second week on tuesday.

I don't think this is maintained anymore, I will attempt to fix it myself. Did you have any luck fixing this?

**Comment by JoeHogan** [comment]
Yeah, sorry folks, i havent looked at this in years and dont really work with sharepoint anymore. It probably deserves a re-write of the utility.

**Comment by shaipetel** [comment]
Sure, I fixed my copy, maybe I can submit my fix to the repo or give it to you to update for future use?

Its simple, a slight confution between getDate and getDay.

This code:
```
for (var i = 0; i < wd.length; i++) { //get first instance of the specified day
    if (arr.indexOf(wd[i]) != -1) {
        if (nd.getDate() > i) nd.setDate(nd.getDate() + (7 - (nd.getDay() - i)));
        else nd.setDate(nd.getDate() + (i - nd.getDay()));
    }
}
```

Needs to be changed to this code:
```
for (var i = 0; i < wd.length; i++) { //get first instance of the specified day
    if (arr.indexOf(wd[i]) != -1) {
        if (nd.getDay() > i) nd.setDate(nd.getDate() + (7 - (nd.getDay() - i)));
        else nd.setDate(nd.getDate() + (i - nd.getDay()));
    }
}
```

So, the check in the if should be if (nd.getDay() > i) instead of getDate.

That seemed to fix it for me, if anyone needs it. @JoeHogan thanks for this utility, its really useful. I don't mind making some fixes on this repo if you grant me permissions.

**Comment by JoeHogan** [comment]
by all means, if you'd like to push some updates, feel free! Sent you an invite

**Comment by shaipetel** [comment]
Please confirm if the issue is now resolved in commit 1988f35
